src/Time/Triggers/NearTimes.hpp

```diff
@@ -47,8 +40,7 @@ enum class Direction { Before, After, Both };
 /// slabs or steps are approximate.
 ///
 /// \see Times
-template <typename TriggerRegistrars = tmpl::list<Registrars::NearTimes>>
-class NearTimes : public Trigger<TriggerRegistrars> {
+class NearTimes : public Trigger {
```

Comment:
The changes to this files and the other factory-creatables are amazing, thanks for doing this @wthrowe.

Skimming the PR I think the design is very reasonable. I don't love the proliferation of the "reaching up to the metavars" pattern to get the list of derived classes, but that's a tradeoff I feel is very much worth it :) I think the metavars are actually the right place to _choose_ the set of factory-creatable classes, possible by just including typelists like `Triggers::logical_triggers`, so it's a good start. We might be able to figure out better ways to pass the information down to code that needs to loop over them later. For instance, I'm thinking we could add a `FactorClasses` tag to the DataBox instead of the full `Metavariables`, or perhaps even `FactoryClasses<Base>` for each base class. Not entirely sure if that's possible. We could also wrap the `factory_classes` alias in the metavars in a struct to give it some documentation and grouping, like this:

```cpp
struct Metavars {
  // ...
  struct factory_creation : tt::ConformsTo<Options::protocols::FactoryCreation> {
    using factory_classes = tmpl::map<...>;
  };
};

// In Options/Protocols/FactoryCreation.hpp:
namespace Options::protocols {
/*!
 * \brief Compile-time information for factory-creation
 *
 * A class conforming to this protocol is placed in the metavariables to provide
 * information on the set of derived classes for each factory-creatable base class.
 * The conforming class must provide the following type aliases:
 *
 * - `factory_classes`: A `tmpl::map` from the base class to the list of derived
 *   classes. List all derived classes that should be factory-creatable, e.g. through
 *   input-file options.
 *
 * Here's an example for a class conforming to this protocol:
 *
 * \snippet Test_FactoryCreation.cpp example
 */
struct FactoryCreation {
  template <typename ConformingType>
  struct test {
    using factory_classes = typename ConformingType::factory_classes;
  };
};
}
```

Of course the docs could be expanded a lot to explain more about factory creation and/or refer to other places in the documentation for details.

In any case, I'm happy to get started merging some of your infrastructure commits.

Reply:
We could certainly wrap this in another struct and put that in the DataBox instead of or in addition to the metavariables.  (We can't put the map itself in the DataBox because it is not serializable.)  The changes for this would be pretty simple, so I can make them if people like that plan.

We are trying to keep the number of tags in the DataBox down, so splitting into one tag per base class is probably not a good idea unless we get some code benefit from doing so.